src/OpenTelemetry.Collector.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -0,0 +1,154 @@
+
+        internal const string SqlDataWriteCommandError = "System.Data.SqlClient.WriteCommandError";
+        internal const string SqlMicrosoftWriteCommandError = "Microsoft.Data.SqlClient.WriteCommandError";
+
+        private readonly PropertyFetcher commandFetcher = new PropertyFetcher("Command");
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher dataSourceFetcher = new PropertyFetcher("DataSource");
+        private readonly PropertyFetcher databaseFetcher = new PropertyFetcher("Database");
+        private readonly PropertyFetcher commandTypeFetcher = new PropertyFetcher("CommandType");
+        private readonly PropertyFetcher commandTextFetcher = new PropertyFetcher("CommandText");
+        private readonly PropertyFetcher exceptionFetcher = new PropertyFetcher("Exception");
+        private readonly SqlClientCollectorOptions options;
+
+        public SqlClientDiagnosticListener(string sourceName, Tracer tracer, SqlClientCollectorOptions options)
+            : base(sourceName, tracer)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+        }
+
+        public override void OnCustom(string name, Activity activity, object payload)
+        {
+            switch (name)
+            {
+                case SqlDataBeforeExecuteCommand:
+                case SqlMicrosoftBeforeExecuteCommand:
+                    {
+                        var command = this.commandFetcher.Fetch(payload);
+
+                        if (command == null)
+                        {
+                            CollectorEventSource.Log.NullPayload($"{nameof(SqlClientDiagnosticListener)}-{name}");
+                            return;
+                        }
+
+                        var connection = this.connectionFetcher.Fetch(command);
+                        var database = this.databaseFetcher.Fetch(connection);
+
+                        this.Tracer.StartActiveSpan((string)database, SpanKind.Client, out var span);
+
+                        if (span.IsRecording)
+                        {
+                            var dataSource = this.dataSourceFetcher.Fetch(connection);
+                            var commandText = this.commandTextFetcher.Fetch(command);
+
+                            span.PutComponentAttribute("sql");
+
+                            span.PutDatabaseTypeAttribute("sql");
+                            span.PutPeerServiceAttribute((string)dataSource);
+                            span.PutDatabaseInstanceAttribute((string)database);
+
+                            if (this.commandTypeFetcher.Fetch(command) is CommandType commandType)
+                            {
+                                span.SetAttribute("db.statementType", commandType.ToString());
```

Comment:
Could you make this a `cosnt string` please.

Reply:
Done